// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package optimization_guide.proto;

import "components/optimization_guide/proto/features/media.proto";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.features.proto";

option java_outer_classname = "ExampleForTestingProto";

message ExampleForTestingMessage {
  bool bool_value = 1;

  string string_value = 2;

  enum ExampleEnum {
    VALUE0 = 0;
    VALUE1 = 1;
  }

  ExampleEnum enum_value = 3;

  Media media = 4;
}

message ExampleForTestingRequest {
  bool bool_value = 1;

  string string_value = 2;

  ExampleForTestingMessage.ExampleEnum enum_value = 3;

  ExampleForTestingMessage nested1 = 4;
  ExampleForTestingMessage nested2 = 5;

  repeated ExampleForTestingMessage repeated_field = 6;
}

message ExampleForTestingResponse {
  bool bool_value = 1;

  string string_value = 2;

  enum ExampleEnum {
    VALUE0 = 0;
    VALUE1 = 1;
  }

  ExampleEnum enum_value = 3;
}

message ExampleForTestingData {
  ExampleForTestingRequest request = 1;
  ExampleForTestingResponse response = 2;
}
